service/routes.py

```diff
@@ -92,11 +92,10 @@ def list_orders():
     orders = []
 
     # Process the query string if any
-    name = request.args.get("name")
-    if name:
-        orders = Order.find_by_name(name)
-    else:
-        orders = Order.all()
+    customer_name = request.args.get("customer_name")
+    product_name = request.args.get("product_name")
+
+    orders = Order.find_by_criteria(customer_name, product_name)
```

Comment:
I think you're missing the case where if we have none of those we do Orders.all(). And will this cover the part where we have either 1 of those (as we might not query both at the same time)?

Reply:
In find_by_criteria(), if "customer_name" and "product_name" are both None, it will do query.all() which would equal to Order.all().
It allows us to add any number of query conditions.

Reply:
> I think you're missing the case where if we have none of those we do Orders.all(). And will this cover the part where we have either 1 of those (as we might not query both at the same time)?

In find_by_criteria(), if "customer_name" and "product_name" are both None, it will do query.all() which would equal to Order.all().
It allows us to add any number of query conditions.